browser/extensions/api/brave_rewards_api.cc

```diff
@@ -1221,20 +1207,16 @@ ExtensionFunction::ResponseAction BraveRewardsGetPrefsFunction::Run() {
 void BraveRewardsGetPrefsFunction::GetAutoContributePropertiesCallback(
     ledger::type::AutoContributePropertiesPtr properties) {
   base::Value prefs(base::Value::Type::DICTIONARY);
+
   prefs.SetBoolKey("autoContributeEnabled", properties->enabled_contribute);
   prefs.SetDoubleKey("autoContributeAmount", properties->amount);
 
-  auto* ads_service = AdsServiceFactory::GetForProfile(
-      Profile::FromBrowserContext(browser_context()));
-
-  if (ads_service) {
-    prefs.SetBoolKey("adsEnabled", ads_service->IsEnabled());
-    prefs.SetDoubleKey("adsPerHour",
-                       static_cast<double>(ads_service->GetAdsPerHour()));
-  } else {
-    prefs.SetBoolKey("adsEnabled", false);
-    prefs.SetDoubleKey("adsPerHour", 0);
-  }
+  Profile* profile = Profile::FromBrowserContext(browser_context());
```

Comment:
this stuff shouldn't be here at all, it should be part of AutoContributePropertiesPtr or returned in some other way by the callback. The extension function should just be doing type conversion (which actually can be handled automatically by mojo). As a short-term fix I would add `ads_enabled` as a second parameter for this method

Reply:
sorry, I take that back after seeing https://github.com/brave/brave-core/pull/9810/files#diff-26a767c397ce1781419dc9c2a14c5258afbc8d6cb1cf906482780f844cf05806R1243
`AutoContributeProperties` should have ads_enabled as a property

Reply:
`ads_enabled` and `ads_per_hour` are not related to auto contribute. For this PR I am trying to make as few logic changes as possible.